java/videostore/src/Customer.java

```diff
@@ -1,67 +1,81 @@
-		rentals.addElement (rental);
-	}
-
-	public String getName () {
-		return name;
-	}
-
-	public String statement () {
-		double 				totalAmount 			= 0;
-		int					frequentRenterPoints 	= 0;
-		Enumeration 		rentals 				= this.rentals.elements ();
-		String 				result 					= "Rental Record for " + getName () + "\n";
-
-		while (rentals.hasMoreElements ()) {
-			double 		thisAmount = 0;
-			Rental 		each = (Rental)rentals.nextElement ();
-
-			// determines the amount for each line
-			switch (each.getMovie ().getPriceCode ()) {
-				case Movie.REGULAR:
-					thisAmount += 2;
-					if (each.getDaysRented () > 2)
-						thisAmount += (each.getDaysRented () - 2) * 1.5;
-					break;
-				case Movie.NEW_RELEASE:
-					thisAmount += each.getDaysRented () * 3;
-					break;
-				case Movie.CHILDRENS:
-					thisAmount += 1.5;
-					if (each.getDaysRented () > 3)
-						thisAmount += (each.getDaysRented () - 3) * 1.5;
-					break;
-			}
-
-			frequentRenterPoints++;
-
-			if (each.getMovie ().getPriceCode () == Movie.NEW_RELEASE
-					&& each.getDaysRented () > 1)
-				frequentRenterPoints++;
-
-			result += "\t" + each.getMovie ().getTitle () + "\t"
-								+ String.valueOf (thisAmount) + "\n";
-			totalAmount += thisAmount;
-
-		}
-
-		result += "You owed " + String.valueOf (totalAmount) + "\n";
-		result += "You earned " + String.valueOf (frequentRenterPoints) + " frequent renter points\n";
-
-
-		return result;
-	}
-
-
-	private String name;
-	private Vector rentals = new Vector ();
+import java.util.ArrayList;
+import java.util.List;
+
+public class Customer {
+    private String name;
+    private List<Rental> rentals = new ArrayList<>();
+    private double amountOwed;
+    private int frequentRenterPoints;
+
+    public Customer(String name) {
+        this.name = name;
+    }
+
+    public void addRental(Rental rental) {
+        rentals.add(rental);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String statement() {
+        amountOwed = 0;
+        frequentRenterPoints = 0;
+        String result = "Rental Record for " + getName() + "\n";
+
+        for (Rental eachMovie : rentals) {
+            double currentAmount = 0;
+
+            amountOwed += getAmountAccrued(eachMovie, currentAmount);
+
+            getFrequentRenterPoints(eachMovie);
+
+            result += "\t" + eachMovie.getMovie().getTitle() + "\t" + String.valueOf(currentAmount) + "\n";
+        }
+
+        result += "You owed " + String.valueOf(amountOwed) + "\n";
+        result += "You earned " + String.valueOf(frequentRenterPoints) + " frequent renter points\n";
+
+
+        return result;
+    }
+
+    public double getAmountOwed() {
+        return amountOwed;
+    }
+
+    public int getFrequentRenterPoints() {
+        return frequentRenterPoints;
+    }
+
+    private double getAmountAccrued(Rental eachMovie, double currentAmount) {
+        return amountAccruedForRentals(currentAmount, eachMovie);

```

Comment:
This is interesting!
The only thing that `getAmountAccrued(...)` does is flip the arguments for `amountAccruedForRentals(...)`.
Doe you see any other value this method brings?
Otherwise I'd suggestion lining up the arguments and removing `getAmountAccrued(...)`

Reply:
Oh dear, well spotted. Thank you, this now fixed.